components/TemplateGrid/TemplateGridWrapper.vue

```diff
@@ -3,93 +3,120 @@
  * See LICENSE for license details.
  */
 <template>
-    <div class="category-tree-wrapper">
-        <TemplateGridHeader
-            :style="gridStyles"
-            :columns="columns" />
+    <div class="template-grid-wrapper">
+        <slot name="gridHeader" />
         <TemplateGridContainer
             :columns="columns"
-            :rows="rowsCount"
+            :rows="rows"
             :rows-height="rowsHeight"
-            :tree-data="filteredTreeData"
-            @toggleItem="(item) => toggleItem(item)">
+            :grid-data="filteredGridData"
+            :is-dragging-enabled="isDraggingEnabled"
+            @toggleItem="(item) => toggleItem(item)"
+            @methodAfterDrop="(data) => methodAfterDrop(data)"
+            @methodAfterRemove="(data) => methodAfterRemove(data)">
             <TemplateGridPresentationLayer
                 :style="gridStyles"
                 :columns="columns"
-                :rows="rowsCount" />
+                :rows="rows" />
             <TemplateGridItemsContainer
                 :style="gridStyles">
                 <TemplateGridItemArea
-                    v-for="item in filteredTreeData"
+                    v-for="item in filteredGridData"
                     :key="item.id"
                     :item="item"
                     :columns="columns"
-                    :rows-height="rowsHeight">
+                    :rows-height="rowsHeight"
+                    :grid-gap="gridGap">
                     <TemplateGridGhostItem
                         v-if="item.ghost" />
-                    <CategoryTreeItem
+                    <slot
                         v-else
-                        :number-of-children="getChildrenLengthById(item.id)"
-                        :is-expanded="getExpandStateById(item.id)"
-                        :item-name="item.name || item.code"
-                        @toggleItem="toggleItem(item)" />
+                        name="gridItem"
+                        :item="item"
+                        :toggle-item-method="toggleItem"
```

Comment:
passing a method is not good idea, use emit from child to parent, communication between parent -> child is created via props